src/context/AirDBContext.tsx

```diff
@@ -71,6 +81,129 @@ export const AirDBProvider = React.memo((props: any) => {
     )
   }
 
+  const handleLogin = ({ password, username, setToggled }: any) => {
+    const isMatching =
+      users.filter(
+        (user: { fields: { password: string; username: string } }) =>
+          user.fields.password === password && user.fields.username === username
+      ).length > 0
+    const userId = (
+      users.filter(
+        (user: { fields: { username: string } }) =>
+          user.fields.username === username
+      )[0] || {
+        id: ''
+      }
+    ).id
+
+    if (isMatching) {
+      setToggled(false)
+      window.localStorage.setItem('username', username)
+      return updateAirDB('Testy - Users', userId, {
+        active: 'true'
+      })
+    } else {
+      throw new Error('INCORRECT PASSWORD')
+    }
+  }
+
+  const handleAddQuestion = ({ options, question }: any) => {
+    const userid = (
+      users.filter(
+        (user: { fields: { username: string } }) =>
+          user.fields.username === window.localStorage.getItem('username')
+      )[0] || { fields: { ID: '' } }
```

Comment:
I like the default value. I would recommend making a separate function at least for getting the current user id. As further step (only if you want), you could also have a `Storage` class or something that just handles the details of where the user gets saved.

Reply:
What do you mean by the `Storage` class? I don’t understand.

Reply:
1. Go ahead and make a function for getting the user id.
2. If you wanted to not hard-wire it to the localStorage (to allow future options), you could just tell some new class called Storage to "saveUsername" or something like that.